app/src/main/java/com/eveningoutpost/dexdrip/StopSensor.java

```diff
@@ -44,9 +44,11 @@ public void addListenerOnButton() {
         button.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
                 Sensor sensor = Sensor.currentSensor();
-                sensor.stopped_at = new Date().getTime();
-                Log.w("NEW SENSOR", "Sensor stopped at " + sensor.stopped_at);
-                sensor.save();
+                if(sensor != null) {
```

Comment:
Well, I have started and stopped so many sensors, that I was able to stop a sensor twice, and then it crashed...